corgidrp/pump_trap_calibration.py

```diff
@@ -0,0 +1,3281 @@
+        'cap': cap1 is either the probability of capture
+        (if trap_fit_const() used) or tauc (capture time constant) if
+        trap_fit() used. cap1_err is the error from fitting.  max_amp1 is the
+        maximum amplitude of the dipole from the curve fit for that pixel.
+        Similarly for cap2, cap2_err, and max_amp2, and there can be a 3rd set
+        of parameters if a trap sub-electrode location was determined using
+        3 schemes.  This data may be useful for future analysis.
+        'E': energy level.  In eV.
+        'sig_E': standard deviation error of energy level.  In eV.
+        'cs': cross section for holes.  In cm^2.
+        'sig_cs': standard deviation error of cross section for holes.
+        In cm^2.
+        'Rsq': adjusted R^2 for the tau vs temperature fit that was done to
+        obtain cs.
+        'tau at input T': tau (in seconds) evaluated at desired temperature of
+        Roman EMCCD, input_T.
+        'sig_tau at input T': standard deviation error of tau at desired
+        temperature of Roman EMCCD, input_T.  Found by propagating error by
+        utilizing 'sig_cs' and 'sig_E'.
+
+    We will recode the string parts of that dictionary specifying the 
+    sub-electrode location for a given pixel to a number code. An example of such 
+    a string is 'RHSel2', which denotes the right-hand side of electrode 2. It can 
+    be 'RHS', 'LHS', or 'CEN' for electrodes 1 through 4. So that can be 
+    converted to a number code (1: 'LHS', 2: 'CEN', 3: 'RHS'), so that 'RHSel2' 
+    would be 32. 
+
+    Args: 
+        trap_dict (dict): A dictionary output by tpump_analysis
+        input_dataset (list): A list of corgidrp.data.TPumpData objects.
+    
+    Returns: 
+        trap_cal (corgidrp.data.TrapCalibration): A trap calibration file. 
+    '''
+    
+    n_traps = len(trap_dict)
+
+    electrode_dict = {"LHS": 10, 
+                      "CEN": 20,
+                      "RHS": 30}
+
+    #Create the main array of traps and their properties, and fill it in
+    trap_cal_array = np.zeros([n_traps,10])
+    for i,key in enumerate(trap_dict.keys()): 
+        key_list = list(key)
+
+        #The first two elements should be the row and column of the trap
+        trap_cal_array[i,0] = key_list[0][0]
+        trap_cal_array[i,1] = key_list[0][1]
+
+        #Convert the electrode_string into a number (described in the docstring)
+        split_electrode_str = key_list[1].split("el")
+        trap_cal_array[i,2] = electrode_dict[split_electrode_str[0]] + \
+                                int(split_electrode_str[1])
+
+        #Which trap is this at this location? 
+        trap_cal_array[i,3] = key_list[2]
+        
+        #Now extract the dictionary entry: 
+        dict_entry = trap_dict[key]
+
+        #Grab the first time constant 
+        #TODO: It seems like there could be three of these. How do we choose which one? 
```

Comment:
True, when I suggested keeping the capture time constant, I kind of forgot there is in general more than one!  But I think this is fine given the upcoming deadline.  I can write up an issue about an enhancement to this step function thaedt includes a slot for the 2nd and 3rd capture time constant which is 0 if non-existent (and a trap is identified by at most 3 of these).  Same goes for max amplitude in line 3258.  And I'll include info about the inclusion of an `err_hdr` in the future as well.

Reply:
Issue made!